programs/address-lookup-table-tests/benches/address_lookup_table.rs

```diff
@@ -0,0 +1,267 @@
+        }
+    }
+
+    fn prep_create_lookup_table(&mut self) {
+        self.instruction_accounts = vec![
+            AccountMeta::new(self.lookup_table_address, false),
+            AccountMeta::new_readonly(self.authority_address, false),
+            AccountMeta::new(self.payer_address, true),
+            AccountMeta::new_readonly(system_program::id(), false),
+        ];
+
+        self.instruction_data = bincode::serialize(&ProgramInstruction::CreateLookupTable {
+            recent_slot: self.recent_slot,
+            bump_seed: self.bump_seed,
+        })
+        .unwrap();
+    }
+
+    fn exec_create_lookup_table(&mut self) {
+        self.prep_create_lookup_table();
+        let accounts = self.run();
+
+        let lookup_table_account = accounts[0].clone();
+        self.transaction_accounts[0] = (self.lookup_table_address, lookup_table_account);
+    }
+
+    fn prep_extend_lookup_table(&mut self) {
+        self.instruction_accounts = vec![
+            AccountMeta::new(self.lookup_table_address, false),
+            AccountMeta::new_readonly(self.authority_address, true),
+            AccountMeta::new(self.payer_address, true),
+            AccountMeta::new_readonly(system_program::id(), false),
+        ];
+
+        // extend reasonable number of addresses at a time, so bench loop won't
+        // add more than LOOKUP_TABLE_MAX_ADDRESSES addresses to lookup_table
+        let new_addresses: Vec<_> = (0..16).map(|_| Pubkey::new_unique()).collect();
+        self.instruction_data =
+            bincode::serialize(&ProgramInstruction::ExtendLookupTable { new_addresses }).unwrap();
+    }
+
+    fn exec_extend_lookup_table(&mut self) {
+        self.prep_extend_lookup_table();
+        let accounts = self.run();
+
+        let lookup_table_account = accounts[0].clone();
+        self.transaction_accounts[0] = (self.lookup_table_address, lookup_table_account);
+    }
+
+    fn prep_freeze_lookup_table(&mut self) {
+        self.instruction_accounts = vec![
+            AccountMeta::new(self.lookup_table_address, false),
+            AccountMeta::new(self.authority_address, true),
+        ];
+
+        self.instruction_data = bincode::serialize(&ProgramInstruction::FreezeLookupTable).unwrap();
+    }
+
+    fn prep_deactivate_lookup_table(&mut self) {
+        self.instruction_accounts = vec![
+            AccountMeta::new(self.lookup_table_address, false),
+            AccountMeta::new(self.authority_address, true),
+        ];
+
+        self.instruction_data =
+            bincode::serialize(&ProgramInstruction::DeactivateLookupTable).unwrap();
+    }
+
+    fn exec_deactivate_lookup_table(&mut self) {
+        self.prep_deactivate_lookup_table();
+        let accounts = self.run();
+
+        let lookup_table_account = accounts[0].clone();
+        self.transaction_accounts[0] = (self.lookup_table_address, lookup_table_account);
+        // advance clock after deactivating
+        self.transaction_accounts[6] = (
```

Comment:
using numbers to index accounts rather than at least named constant makes it fragile in case of code modifications.

Reply:
yea agree, thanks for good suggestion! can refactor it separately.